packages/text-annotator/src/SelectionHandler.ts

```diff
@@ -66,12 +64,38 @@ export const SelectionHandler = (
       return;
     }
 
-    // Chrome/iOS does not reliably fire the 'selectstart' event!
+
     const timeDifference = evt.timeStamp - (lastPointerDown?.timeStamp || evt.timeStamp);
-    if (timeDifference < 1000 && !currentTarget)
+
+    if (timeDifference < 1000 && !currentTarget) {
+
+      // Chrome/iOS does not reliably fire the 'selectstart' event!
       onSelectStart(lastPointerDown);
 
-    if (sel.isCollapsed || !isLeftClick || !currentTarget) return;
+    } else if (sel.isCollapsed && timeDifference < CLICK_TIMEOUT) {
+
+      /*
+       Firefox doesn't fire the 'selectstart' when user clicks
+       over the text, which collapses the selection
+      */
+      onSelectStart(lastPointerDown);
+
+    }
```

Comment:
Those lines unify Firefox's handling of the click over the textual content with the rest of the browsers. It's _the only browser_ that has that doesn't dispatch the `selectstart` when the selection range collapses.
Therefore, when the user clicks somewhere on the screen upon finishing the selection, that new annotation will get falsely deleted [here](https://github.com/recogito/text-annotator-js/pull/140/files#diff-1e072053802159d91da89e331e1fe166da3cb395a3fc6ab3d6317b0510ea17e0R94-R98), because the `currentTarget` will still refer to it. But in Chrome/Safari, the `selectstart` event will be dispatched, and the `currentTarget` will get updated to the one that doesn't exist in the store yet.
Demos - https://github.com/recogito/text-annotator-js/pull/140#issuecomment-2307515070